### ISSUE TITLE
Getting "unclosed connection" errors when running the demo

### ISSUE THREAD

**Issue by elemoine** [peopledoc/procrastinate#157]
We get "unclosed connection" errors when running `python -m procrastinate_demo` to schedule tasks.

```
$ python -m procrastinate_demo 
DEBUG:asyncio:Using selector: EpollSelector
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
INFO:procrastinate.jobs:Deferring job
/home/elemoine/.virtualenvs/procrastinate/lib/python3.8/site-packages/aiopg/connection.py:513: ResourceWarning: Unclosed connection <aiopg.connection::Connection isexecuting=False, closed=True, echo=False, cursor=<aiopg.cursor::Cursor name=None, closed=True>>
ERROR:asyncio:Unclosed connection
connection: <aiopg.connection::Connection isexecuting=False, closed=True, echo=False, cursor=<aiopg.cursor::Cursor name=None, closed=True>>
```

**Comment by ewjoachim** [comment]
That line should take care of that, though... 
https://github.com/peopledoc/procrastinate/blob/master/procrastinate/cli.py#L109

**Comment by elemoine** [comment]
The cli is not used when we do `python -m procrastimate_demo` as indicated in the contributing file.

**Comment by ewjoachim** [comment]
Then maybe we should either:
- add "try: finally:" blocks in app's worker (and maybe migrator and health checker)
- Add a decorator that does this

I feel that when interacting with the app, the app's responsible for requesting the connection to open, so it should be also the one that closes it.

(maybe transparently opening the connection was not a great idea)

Point being: I'm not sure relying on the GC to close the connection is my favorite approach here :/

**Comment by elemoine** [comment]
> Then maybe we should either:
> 
>    * add "try: finally:" blocks in app's worker (and maybe migrator and health checker)
>    * Add a decorator that does this

I don't understand that part. This issue is about the defer side, so I don't understand why you're talking about the worker.

> I feel that when interacting with the app, the app's responsible for requesting the connection to open, so it should be also the one that closes it.

What does `app` refer to in the above statement? The `procrastinate.App` object, or the user application? I am not sure…

As I see it, if we don't want to rely on the GC we need to make `App.close_connection` public (or `PostgresConnector.close_connection` maybe), making it appear in the reference API. And the user would be responsible for calling `app.close_connection` when done with her Procrastinate application.

**Comment by ewjoachim** [comment]
> I don't understand that part. This issue is about the defer side, so I don't understand why you're talking about the worker.

At the time of writing this comment I had not realized the problem was at the defer side.

> I feel that when interacting with the app, the app's responsible for requesting the connection to open, so it should be also the one that closes it.

Procrastinate App. This sentence was later rephrased as "User did not request to open the connection, they should not have to close it." and since then, we've expressed that we're on the same line on that one, just need to find the right abstraction to make it simple for them (and ideally us)

**Comment by ewjoachim** [comment]
Closed by #158
